### PR TITLE
request and response have the same domain schema

### LINKED ISSUE #2046: EPP: responses must reference the same schema version that was used in valid request
related to https://github.com/internetee/registry/issues/2030

with support for multiple schema file versions we need also make sure we are responding with the same schema file paths that were used in valid request

### PR DESCRIPTION
**Pull request by OlegPhenomenon**
Close #2046

### CODE REVIEW

File: app/controllers/epp/domains_controller.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class DomainsController < BaseController
     before_action :find_domain, only: %i[info renew update transfer delete]
     before_action :find_password, only: %i[info update transfer delete]
     before_action :set_paper_trail_whodunnit
+    before_action :parsing_schemas_prefix_and_version
```
Comment (yulgolem) [inline app/controllers/epp/domains_controller.rb:7]: Method name should be without `-ing` suffix, so just `parse_schema_prefix_and_version` will do.

File: app/controllers/epp/domains_controller.rb [hunk 2]
```diff
@@ -22,7 +23,7 @@ def info
       when 'all'
         @nameservers = @domain.nameservers.sort
       end
-
+
```
Comment (yulgolem) [inline app/controllers/epp/domains_controller.rb:26]: Please, remove these whitespaces.

File: app/views/epp/domains/create.xml.builder [hunk 3]
```diff
@@ -5,7 +5,8 @@ xml.epp_head do
     end
 
     xml.resData do
-      xml.tag!('domain:creData', 'xmlns:domain' => Xsd::Schema.filename(for_prefix: 'domain-ee')) do
+      xml.tag!('domain:creData',
+               'xmlns:domain' => Xsd::Schema.filename(for_prefix: @schema_prefix)) do
```
Comment (yulgolem) [inline app/views/epp/domains/create.xml.builder:9]: And what about version? As far as I remember `Xsd::Schema.filename(for_prefix: <some_prefix>))` still returns latest version of a filename.

**Review by yulgolem — CHANGES_REQUESTED** [review]

**Comment by yulgolem** [comment]
I suppose we still need to create a test to check if we are returning not the latest schema version, but the version from the request if that's supported. ATM I can't find any.

**Comment by OlegPhenomenon** [comment]
> I suppose we still need to create a test to check if we are returning not the latest schema version, but the version from the request if that's supported. ATM I can't find any.

test/integration/epp/doman/info/base_test.rb

test_returns_valid_response_if_schema_version_is_previous

**Comment by yulgolem** [comment]
> > I suppose we still need to create a test to check if we are returning not the latest schema version, but the version from the request if that's supported. ATM I can't find any.
> 
> test/integration/epp/doman/info/base_test.rb
> 
> test_returns_valid_response_if_schema_version_is_previous

That's it! We are getting not the response we need, but a last one - since we only got domain-eis-1.0.xsd .
Try to add  domain-eis-1.1.xsd and check if the response persists.

File: app/views/epp/domains/create.xml.builder (hunk 3 above)
Comment (OlegPhenomenon) [reply app/views/epp/domains/create.xml.builder:9]: Related #2058

**Comment by OlegPhenomenon** [comment]
> > > I suppose we still need to create a test to check if we are returning not the latest schema version, but the version from the request if that's supported. ATM I can't find any.
> > 
> > 
> > test/integration/epp/doman/info/base_test.rb
> > test_returns_valid_response_if_schema_version_is_previous
> 
> That's it! We are getting not the response we need, but a last one - since we only got domain-eis-1.0.xsd .
> Try to add domain-eis-1.1.xsd and check if the response persists.

Related #2058

**Pull request merged**